crates/crabe_math/src/shape/rectangle.rs

```diff
@@ -1,16 +1,49 @@
-use nalgebra::Point2;
+use nalgebra::{Point2};
 use serde::Serialize;
 
 /// A rectangle in 2D space, defined by a width, a height, and a position.
-///
+/// 
 /// Note that the `width` and `height` fields should have the same units of
 /// measurement as the coordinates of the `position` field.
+///
+/// The base (in French: repère) is defined as follows :
+/// ```txt
+/// * -- -- -- >
+/// |          x
+/// |
+/// |
+/// v y
+/// ```
 #[derive(Clone, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
 pub struct Rectangle {
     /// The width of the rectangle.
     pub width: f64,
     /// The height of the rectangle.
     pub height: f64,
     /// The position of the rectangle's top-left corner.
-    pub position: Point2<f64>,
+    pub top_left: Point2<f64>,
```

Comment:
You can use memoisation @benjaminchew4 purpose to use this crate to have this [crate](https://crates.io/crates/memoize).

Reply:
![image](https://github.com/NAMeC-SSL/CRAbE/assets/24627625/60d76c0e-015f-439b-8ce4-a5e122b16767)
I can't use this crate for method. Technically, I could just create a function `make_rectangle` inside `rectangle.rs` and add the memoize attribute to it, but I don't think it looks like very clean code.

Reply:
I mean I think we can take care of this in particular in another PR instead of this one. Looks like too much responsibility on a single PR

Reply:
The overhead of memoization would likely outweigh the benefits, as it involves a HashMap lookup. However, storing all the points of a shape decreases readability and increases the likelihood of bugs. For rectangles, I think it would be better to have a struct containing the origin, the width and the height. Points could be accessed with methods, e.g:
```rust
pub fn top_right(&self) -> Point2 {
  Point2::new(self.origin.x + self.width, self.origin.y)
}
```

Reply:
this has been taken care of by [ad7f3c7](https://github.com/NAMeC-SSL/CRAbE/pull/93/commits/ad7f3c72b001c426f8a1df1d4cd91ffedc625cbc)